File: Data Structures & Algorithms/length-of-last-word/submission-6.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        l = len(s) - 1
        res = 0
        if len(s) == 1:
            return 1

        while l >= 0:
            if s[l].isalnum():                
                res += 1
            elif res > 0 and not s[l].isalnum():
                return res
            l -= 1
        return res

File: Data Structures & Algorithms/top-k-elements-in-list/submission-6.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {        
        List<Integer>[] freq = new List[nums.length + 1];
        Map<Integer, Integer> hash = new HashMap<>();

        for (int i=0; i<freq.length; i++){
            freq[i] = new ArrayList<>();
        }

        for (int n: nums){
            hash.put(n, hash.getOrDefault(n, 0) + 1);
        }

        for (Map.Entry<Integer, Integer> entry: hash.entrySet()){
            freq[entry.getValue()].add(entry.getKey());            
        }

        int[] res = new int[k];
        int index = 0;
        for (int i = freq.length - 1; i>0; i --){
            for (int n: freq[i]){
                res[index++] = n;
                if (index == k){
                    return res;
                }
            }
        }
        return res;
    }
}
